pkg/engine/game.go

```diff
@@ -202,19 +202,19 @@ func (g *Game) Cut(src, dst PlayerID, i int) error {
 		// nothing happens
 	}
 
-	// if we've reached the end of the round, deal a new hand
-	cuts := len(g.Cuts)
-	if cuts > 0 && cuts%n == 0 {
-		g.deal()
-		g.Round++
-	}
-
```

Comment:
Without the g.Round++ on line 209, the else if on line 215 does not detect game over at round 4 and will continue for an extra cut into round 5.

Reply:
b9443a3 should fix this